### ISSUE TITLE
It would be great if we generate CSS and JS based on active elements on page. [ Suggetion ]

### ISSUE THREAD

**Issue by sandeshjangam** [pojome/elementor#868]
## Description

Lets say I am using only accordion on page. But now at this moment elemetor load whole JS and CSS file which comes with CSS of other elements also. This increase code bloat.

As currently you generating css file post. If we generate css based on elements. Would be great.

Example - I have widget - Accordion

**# Module Structure**

 **#Accordion.php - Widget File**
  - All controls html etc here,

 **#Frontend.css - Common css for all instance of accordion module.** 
  - This css contain common css of all accordion instance. on save we will append this this css in you current external generated css through your way. It will append only once.

 **#Frontend.css.php - Dynamic CSS based on settings** 
  - In this file we can access $settings variable and bases on setting will write css. Then on save we will append this this css in you current external generated css through your way. It will append as per module instance.

**#Frontend.js - Common js for all instance of accordion module**
  - We will create external generated js as same as external css way.  This js contain object of module. on save we will append this js in external generated js.  It will append only once.

 **#Frontend.js.php - Dynamic JS based on settings**
 - In this file we can access $settings variable and bases on setting we will initiate js object and pass required $settings value which has written in Frontend.js. Then on save we will append this this js in external generated js. It will append as per module instance. 

It will reduced code bloat. It will generate only required css and js. It will also reduced page load time.

**Comment by arielk** [comment]
Thanks for your feedback!

We are definitely working on this improvement for the next versions, for now I merged this issue with another issue #736  in the same subject so that our Dev staff could deal with it more efficiently. So please let's continue the conversation there.
